lib/server.js

```diff
@@ -145,51 +150,68 @@ module.exports = function createServer(config, log) {
             }
 
             var retryAfter = [blockEmail, blockIpEmail, blockIp].reduce(max)
+            var block = retryAfter > 0
+            var suspect = false
 
-            var allowedIP = ip in allowedIPs.ips
-            var allowedEmailDomain = allowedEmailDomains.isAllowed(email)
-
-            if (retryAfter > 0) {
-              if (allowedIP) {
-                retryAfter = 0
-              }
-              if (allowedEmailDomain) {
-                retryAfter = 0
-              }
+            if (requestChecks.treatEveryoneWithSuspicion) {
+              suspect = true
             }
+            // The private branch puts some additional request checks here.
+            // We just use the variables so that eslint doesn't complain about them.
+            payload || headers
+
             var canUnblock = emailRecord.canUnblock()
 
             // IP's that are in blocklist should be blocked
             // and not return a retryAfter because it is not known
             // when they would be removed from blocklist
-            if (config.ipBlocklist.enable && blockListManager.contains(ip) && !allowedIP && !allowedEmailDomain) {
+            if (config.ipBlocklist.enable && blockListManager.contains(ip)) {
               if (!config.ipBlocklist.logOnly) {
-
-                // When blocklist is enabled, explicitly skip setting records.
-                // Requests issued while on blocklist will not count towards
-                // the rate limit.
```

Comment:
I'm not sure there's too much value in avoiding the write here, and removing the `return` means that there's now a single place where we return from this function, and can check that we include all the relevant fields in the response.  I'm open to putting it back in though.